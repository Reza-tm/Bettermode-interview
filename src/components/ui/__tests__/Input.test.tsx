import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Input } from "@ui";
import userEvent from "@testing-library/user-event";

describe("Input", () => {
  it("renders correctly", () => {
    const result = render(<Input label="Username" />);
    expect(screen.getByText("Username")).toBeInTheDocument();
    expect(result).matchSnapshot();
  });

  it("renders error message", () => {
    render(<Input error="This field is required" />);
    expect(screen.getByText("This field is required")).toBeInTheDocument();
  });

  it("error styles on error", () => {
    render(<Input error="Error" />);
    const input = screen.getByRole("textbox");
    expect(input).toHaveClass("ring-red-500");
  });

  it("handles input changes", async () => {
    const user = userEvent.setup();
    render(<Input />);
    const input = screen.getByRole("textbox");
    await user.type(input, "test");
    expect(input).toHaveValue("test");
  });
});
