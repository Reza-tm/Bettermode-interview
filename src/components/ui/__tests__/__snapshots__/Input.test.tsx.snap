// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="text-content-on-background"
        >
          Username
        </span>
        <input
          class="w-full appearance-none rounded-md border border-line bg-transparent px-3 py-2 text-content-on-background focus-within:ring-action-primary ring-focused ring-offset-0 focus-within:border-action-neutral focus-within:ring-1 focus:outline-none transition duration-200"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-2"
    >
      <span
        class="text-content-on-background"
      >
        Username
      </span>
      <input
        class="w-full appearance-none rounded-md border border-line bg-transparent px-3 py-2 text-content-on-background focus-within:ring-action-primary ring-focused ring-offset-0 focus-within:border-action-neutral focus-within:ring-1 focus:outline-none transition duration-200"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
